### ISSUE TITLE
CVE-2023-5072 - Denial of Service in JSON-Java versions up to and including 20230618.

### ISSUE THREAD

**Issue by elrob** [stleary/JSON-java#789]
https://nvd.nist.gov/vuln/detail/CVE-2023-5072

https://github.com/advisories/GHSA-rm7j-f5g5-27vv

**Comment by johnjaylward** [comment]
Relates to #758, #759, #771, #772

**Comment by stleary** [comment]
@elrob Thanks for letting us know. As @johnjaylward posted, it seems to me that both vulnerabilities have been addressed and it is only necessary to cut a new release. I don't think any additional code changes are needed.

**Comment by TimoBuechert** [comment]
@stleary can you already tell when the new release will be created?

**Comment by stleary** [comment]
Yes, the new release will come out later today.

**Comment by Bhavitha-Chava** [comment]
Hi @stleary, Could you pls let me know when we can expect the release? Is there a lower version that we can use to avoid hitting this vulnerability

**Comment by stleary** [comment]
Release `20231013` is available, and is the earliest version that fixes the vulnerability. Have patience, it can take some time before it appears in the Maven repo.

**Comment by lhazlewood** [comment]
@stleary was it strictly necessary to require JDK 8 for this release?  `20230618` was still JDK 7 compatible, but `20231013` is not.

**Comment by stleary** [comment]
@lhazlewood It was not strictly necessary. 
How important is this to you? 
What version of Java do you use?
Can you upgrade to Java 8?

**Comment by stleary** [comment]
@elrob @TimoBuechert Release `20231013` is now available in the Maven public repository.

**Comment by lhazlewood** [comment]
@stleary I maintain a library that is used with both Java and Android, and we support JDK 7 (at the moment) mostly for Android's purposes (which uses `org.json` APIs).  That requirement for us is going away shortly however, and I did find a workaround using maven profiles to default to `20231013`, with an override to `20230618` when compiling on JDK 7 for the time being.

**Comment by PayBas** [comment]
Haven't tested it thoroughly yet, but our main issue is that `20231013` seems to be included in the vulnerable versions in our Nexus Lifecycle tool:
![image](https://github.com/stleary/JSON-java/assets/3015760/a95ca2ab-10ad-4c34-8eba-2dfdbd028229)

Not sure if it's an administration error, or if they feel that `20231013` doesn't actually fixes the vulnerability.

I'll try to contact SonaType to see whats up.

**Comment by lhazlewood** [comment]
If there is to be another release for this, would it be possible to build with JDK 7?  My maven profile workaround didn't work as I expected.  If not, I'll understand, but I thought I'd ask in case it wasn't too difficult a request to entertain.

**Comment by PayBas** [comment]
@stleary it seems to have worked.

https://ossindex.sonatype.org/component/pkg:maven/org.json/json@20231013

> This version of json has no known vulnerabilities! 🎉

**Comment by stleary** [comment]
The CVE still appears to be under analysis, but hopefully it will be cleared soon, too.
https://nvd.nist.gov/vuln/detail/CVE-2023-5072

**Comment by stleary** [comment]
@lhazlewood Yes, this can be done. Will the same code in a different repo work for you?

**Comment by lhazlewood** [comment]
@stleary how do you mean different repo?

**Comment by stleary** [comment]
@lhazlewood It has not been decided yet. Might be a different repo that is published to Maven and tracks JSON-Java but is Java 7 compatible, or #741 might be reverted, which could get complicated. Do you have any thoughts or concerns about either option?

**Comment by johnjaylward** [comment]
> @lhazlewood It has not been decided yet. Might be a different repo that is published to Maven and tracks JSON-Java but is Java 7 compatible, or #741 might be reverted, which could get complicated. Do you have any thoughts or concerns about either option?

I think it would be best if we could make a `java6` branch based off the tag for release 20130618. Then we would backport any PRs that are related to a CSV (like #759 and #772). The `version` for the branch would stay at a base value of `20130618`, but we could point release it for security fixes like `20130618.001` to issue a fix for this CSV.

The `master` branch could then stay as supporting Java 8+ while only minimal required changes happen on the `java6` branch for security issues.

Does that work for you @stleary ?

**Comment by lhazlewood** [comment]
@stleary @johnjaylward that sounds like a nice option if possible!

**Comment by johnjaylward** [comment]
@stleary , here is an example of what the PR would look like if we created the `java6` branch in this repo

https://github.com/johnjaylward/JSON-java/pull/2

**Comment by stleary** [comment]
@johnjaylward Your idea sounds like a good approach and probably the least disruptive of the options. What do you think this would look like in the Maven repo?

**Comment by johnjaylward** [comment]
Should add the point releases under the java 8 releases

![Screenshot_20231017_181330_Firefox](https://github.com/stleary/JSON-java/assets/383703/aa48ceae-5a5e-4633-9b76-96657590e09f)

**Comment by nathan454** [comment]
> @stleary , here is an example of what the PR would look like if we created the `java6` branch in this repo
> 
> [johnjaylward#2](https://github.com/johnjaylward/JSON-java/pull/2)

@stleary When can we expect backport for 20230618 ? I think @johnjaylward approach sounds ok

**Comment by stleary** [comment]
@nathan454 Are you working on the same project as @lhazlewood, or is this a new request?

**Comment by nathan454** [comment]
> @nathan454 Are you working on the same project as @lhazlewood, or is this a new request?

new request
I'm also using this version and I support a library that needs to maintain support for java 7

**Comment by alenamor7** [comment]
Hi @johnjaylward,
please correct me if I'm wrong: CVE-2023-5072 is fixed in the 20230618.001 version, but is not fixed in the newest 20231013 version? We still have a vulnerablity in the project even though the newest 20231013 version is used. Thanks

**Comment by johnjaylward** [comment]
20231013 should be the fixed version. we decided to NOT point release 20230618 at this time.

**Comment by johnjaylward** [comment]
I don't see an issue with it. Would just need to update the deployment yml to compile to java7

**Comment by stleary** [comment]
Closing due to issue resolved
